go/enclave/storage/enclavedb/block.go

```diff
@@ -56,8 +56,8 @@ func IsCanonicalBlock(ctx context.Context, dbtx *sql.Tx, hash *gethcommon.Hash)
 }
 
 // CheckCanonicalValidity - expensive but useful for debugging races
-func CheckCanonicalValidity(ctx context.Context, dbtx *sql.Tx) error {
-	rows, err := dbtx.QueryContext(ctx, "select count(*), height from batch where is_canonical=true group by height having count(*) >1")
+func CheckCanonicalValidity(ctx context.Context, dbtx *sql.Tx, blockId int64) error {
```

Comment:
we have a bug where we end up with non-canonical batches that are marked as canonical, which breaks the rollup compression/decompression and shows up as some weird xchain panic.

This should help narrow it down